### PR TITLE
Display composer on Now Playing page

### PR DESCRIPTION
**Pull request by whatdoineed2do**
Feature request: https://github.com/ejurgensen/forked-daapd/issues/614

Add `composer`to `queue` table and displaying on `Now Playing` page if `composer` is exposed by `httpd_jsonapi`

composer

### CODE REVIEW

File: src/db_init.h [hunk 1]
```diff
@@ -25,8 +25,8 @@
  * version of the database? If yes, then it is a minor upgrade, if no, then it
  * is a major upgrade. In other words minor version upgrades permit downgrading
  * forked-daapd after the database was upgraded. */
-#define SCHEMA_VERSION_MAJOR 19
-#define SCHEMA_VERSION_MINOR 10
+#define SCHEMA_VERSION_MAJOR 20
+#define SCHEMA_VERSION_MINOR 0
```
[the remark this answers is not in the document]
Comment (ejurgensen) [reply src/db_init.h]: True. For files/playlists there is a column sync check (can't remember where) that means that adding new columns isn't backwards compatible, even though it should be. I think I have fixed this in my pr #595 (not totally sure).

I can't remember if there is also such a sync check for queue. I suggest you in any case make it a minor upgrade. Hopefully I will soon get time to finalise pr #595 and then there will be a major upgrade anyway.

**Comment by chme** [comment]
I don't think, we can rely on the genre to identify if it is a classical music track or not. And the JSON API should always send the data if it is available. 

As a first step, i would add the information to the details dialog (track and queue item). I am planning to refactor the dialog so that it can be called from the now playing page. But it will be better to keep the web interface change separate from this pr.

**Comment by whatdoineed2do** [comment]
Done:  web-src item dropped as well as the `genre == "classical"` check for populate `composer`.

I wonder how we would identify/determine when to show the `composer` field in the UI; I expect that anyone who would tag their classic files for the composer would use this fld but as you point out this fld is not exclusively reserved for classical genre.  Asking people to retag in another location is probably a non-starter...  a config value that specifies the condition on when to display `composer` perhaps?

**Comment by whatdoineed2do** [comment]
@chme - would this 5828992fc (not incl in this PR) be something more to your thinking?  The UI owns/decides whether to show the composer .. on my own build, its show `composer` based on `genre==classical` but this condition (i guess to be expanded) is where you currently have reservations.

```+++ b/web-src/src/pages/PageNowPlaying.vue
@@ -4,6 +4,9 @@
       <div class="container has-text-centered fd-has-margin-top">
         <h1 class="title is-4">
           {{ now_playing.title }}
+          <div class="subtitle is-6" v-show="composer_visible">
+            {{ composer() }}
+          </div>
         </h1>
         <h2 class="title is-6">
           {{ now_playing.artist }}
@@ -64,7 +67,8 @@ export default {
     return {
       item_progress_ms: 0,
       interval_id: 0,
-      artwork_visible: false
+      artwork_visible: false,
+      composer_visible: false
     }
   },
 
@@ -117,6 +121,15 @@ export default {
 
     artwork_error: function () {
       this.artwork_visible = false
+    },
+
+    composer: function () {
+      if (this.now_playing.composer === undefined || this.now_playing.composer === null || this.now_playing.genre === undefined || this.now_playing.genre === null) {
+        this.composer_visible = false
+      } else {
+        this.composer_visible = (this.now_playing.genre.toLowerCase() === 'classical')
+      }
+      return this.now_playing.composer
     }
   },

```

File: src/db.c [hunk 2]
```diff
@@ -5103,6 +5106,7 @@ queue_enum_fetch(struct query_params *qp, struct db_queue_item *queue_item, int
   queue_item->track = sqlite3_column_int(qp->stmt, 20);
   queue_item->disc = sqlite3_column_int(qp->stmt, 21);
   queue_item->artwork_url = strdup_if((char *)sqlite3_column_text(qp->stmt, 22), keep_item);
+  queue_item->composer = strdup_if((char *)sqlite3_column_text(qp->stmt, 24), keep_item);
```
Comment (chme) [inline src/db.c]: Does this work? Should it not be `23`?

Comment (whatdoineed2do) [reply src/db.c]: column 23 is `queue_version` .  I add the column here and it becomes col 24; the full definition is in `db_init.c`

https://github.com/ejurgensen/forked-daapd/pull/619/files#diff-8fc831350c1c371efb7094030a5b5ff2

I did notice that the previous `queue` table updates took 2 different approaches:
* add the column (see `U_V1903_...`)
* drop the `queue` tbl entirely and recreate (see  `U_V1906_...`)

I went with the first approach;  this was tested upgrading an existing db (resulting in the alter tb and addition of column) as well as having no db file (creates via the `db_init`)

Comment (chme) [reply src/db.c]: Thanks, i missed that the queue version is missing. Doing it with an alter statement is correct. Dropping a table should only be done if an alter table is not possible. 

@ejurgensen imo this is good for merge.

**Comment by ejurgensen** [comment]
Merged, thanks a lot!

**Comment by LordMyschkin** [comment]
I still do not see the composer on "now playing" in the web interface - am I missing sth?
Second, I want to add that genre=classical is good for most but not all cases (e.g. soundtracks - wouldn't it be nice to know  the great piece playing is from Hans Zimmer or Ennio Morricone; the same with musicals. And some ppl discriminate between genre == "opera" vs. genre == "chorus" - you get the idea, I think)

**Comment by whatdoineed2do** [comment]
@chme had some reservations about the conditions in which the `composer` was to be displayed.  As you (@LordMyschkin) also note, there may be other considerations than the one I assumed (which was just show the `composer` when `genre == classical`) and hence this was not included in the merge.

However I have this enabled on my personal branch and you can lift and drop the `htdocs` files from here: https://github.com/whatdoineed2do/forked-daapd/tree/_ray/htdocs or via  this commit: https://github.com/whatdoineed2do/forked-daapd/commit/dbe4a7e4001af7f39ce2da356ea1e43f313df1fb if you really really want this functionality (along with some others).

Obviously the code from my branch is not supported via the main branch.

**Comment by LordMyschkin** [comment]
Thank's a lot, @whatdoineed2do , now using your frontend - works as expected, I love it despite the concerns mentioned above about relying on the genre.

**Pull request merged**